haystack/nodes/preprocessor/splitter.py

```diff
@@ -0,0 +1,755 @@
+                                - "word". The text is split with the `split()` function (as done by the old PreProcessor).
+
+                                Defaults to "word".
+
+        :param nltk_language: If `split_by="sentence"`, the language used by "nltk.tokenize.sent_tokenize", for example "english", or "french".
+                                Mind that some languages have limited support by the tokenizer: for example, it seems incapable to split Chinese text
+                                by word, but it can correctly split it by sentence.
+
+        :param nltk_folder: If `split_by="sentence"`, specifies the path to the folder containing the NTLK `PunktSentenceTokenizer` models,
+                            if loading a model from a local path. Leave empty otherwise.
+
+        :param progress_bar: Whether to show a progress bar.
+
+        :param add_page_number: Add the number of the page a paragraph occurs in to the Document's meta
+                                field `"page"`. Page boundaries are determined by `"\f"' character which is added
+                                in between pages by `PDFToTextConverter`, `TikaConverter`, `ParsrConverter` and
+                                `AzureConverter`.
+        """
+        super().__init__()
+        self._validate_split_params(
+            split_by=split_by,
+            split_regex=split_regex,
+            split_length=split_length,
+            split_overlap=split_overlap,
+            max_chars=max_chars,
+            max_tokens=max_tokens,
+        )
+        try:
+            nltk.data.find("tokenizers/punkt")
+        except LookupError:
+            nltk.download("punkt")
+
+        self.split_by = split_by
+        self.split_regex = split_regex
+        self.max_chars = max_chars
+        self.max_tokens = max_tokens
+        self.progress_bar = progress_bar
+        self.add_page_number = add_page_number
+
+        self.merger = DocumentMerger(
+            separator="",
+            window_size=split_length,
+            window_overlap=split_overlap,
+            max_tokens=max_tokens,
+            retain_page_number=True,
+            realign_headlines=True,
+        )
+
+        self._nltk_language = nltk_language
+        self._nltk_folder = Path(nltk_folder) if nltk_folder else None
+        self.sentence_tokenizer = load_sentence_tokenizer(
+            language=self._nltk_language, tokenizer_model_folder=self._nltk_folder
+        )
+
+        self._tokenizer = None
+        if tokenizer_model or max_tokens:
+            self.tokenizer = tokenizer_model
+
+        if split_by == "token" and not self.tokenizer:
+            raise ValueError(
+                "If you set split_by='token', you must give a value to 'tokenizer_model'. "
+                "Use the same model you're using for your Reader."
+            )
+
+    def _validate_split_params(
+        self,
+        split_by: SplitBy,
+        split_regex: Optional[str],
```

Comment:
I think since this is specified an optional parameter it should have its default value set to `None`. Otherwise, the typing says it is optional, but when calling this function it will complain if `split_regex` is not provided.